hie-plugin-api/Haskell/Ide/Engine/PluginsIdeMonads.hs

```diff
@@ -261,10 +261,13 @@ type FormattingProvider = T.Text -- ^ Text to format
         -> FormattingOptions -- ^ Options for the formatter
         -> IdeM (IdeResult [TextEdit]) -- ^ Result of the formatting or the unchanged text.
 
+newtype PluginId = PluginId T.Text
+  deriving (Show, Read, Eq, Ord)
+instance IsString PluginId where
+  fromString = PluginId . T.pack
+
 data PluginDescriptor =
   PluginDescriptor { pluginId                 :: PluginId
-                   , pluginName               :: T.Text
-                   , pluginDesc               :: T.Text
```

Comment:
I don't see why removing these fields is desirable at all. Surely it is useful to be able to inspect a generic plugin, even if for debugging purposes.

Reply:
These are just vanity names/descriptions, which aren’t accessible to the user at all via LSP. For debugging you’ll still always be able to read the pluginId which is like a string “applyRefact” or “floskell”, which is basically the same as the plugin name